packages/commonwealth/server/farcaster/frames/contest/checkEligibility.tsx

```diff
@@ -1,11 +1,58 @@
 import { Button } from 'frames.js/express';
 import React from 'react';
 import { frames } from '../../config';
-import { CardWithText } from '../../utils';
+import { circleCheckIcon, circleXIcon, fakeApiCall } from '../../utils';
 
 export const checkEligibility = frames(async () => {
+  let eligible: boolean;
+
+  try {
+    await fakeApiCall({ error: 'error' });
+    eligible = true;
+  } catch {
+    eligible = false;
+  }
+
+  const base64Icon = eligible ? circleCheckIcon : circleXIcon;
+  const title = eligible ? 'You are eligible to enter' : 'You are not eligible';
+  const description = eligible
+    ? 'Reply to this cast or quote this frame to be entered into the contest.'
+    : 'In order to enter this contest you must connect an Ethereum wallet to your Farcaster account.';
+
   return {
-    image: CardWithText({ text: 'Eligibility' }),
+    image: (
+      <div
+        style={{
+          backgroundColor: '#2A2432',
+          color: 'white',
+          padding: '40px',
+          display: 'flex',
+          flexDirection: 'column',
+          alignItems: 'center',
+          justifyContent: 'center',
+          width: '100%',
+          height: '100%',
+          textAlign: 'center',
+        }}
```

Comment:
can we inject it via .scss class?

Reply:
Not possible. Eventually this markup becomes a static image so it does not have an access css files like it is in the regular websites. You can read [here](https://vercel.com/docs/functions/og-image-generation) how this works behind the scenes. For now only tailwind is supported but it is in experimental mode.

Reply:
Thanks